lib/flat_search/flats.ex

```diff
@@ -26,9 +48,25 @@ defmodule FlatSearch.Flats do
 
   def get_flats, do: Repo.all(Flat)
 
+  def get_flats_by(params) do
+    params
+    |> Enum.reduce(Flat, fn
+      {_field, none}, query when none in ["", nil] ->
+        query
+
+      {"max_price", max_price}, query ->
+        from q in query, where: q.price + q.additional_price <= ^String.to_integer(max_price)
```

Comment:
Can this be rewritten to the same style as line 61? To keep consistent with the styling inside the function

Reply:
Also, consider moving this whole anonymous function to the separate set of function definitions, and just passing it to the reduce function

Reply:
I've tried at first but I can't compare inside a keyword list, so that's what I came up as a work around.

What do you mean by separate set of function definitions? Another module?

Reply:
```
defp generate_query({"max_price", max_price}, query, do: from q in query, where: q.price + q.additional_price <= ^String.to_integer(max_price)
```
etc

Reply:
I still think this whole anonymous function could be moved to other functions and split into different headers